### ISSUE TITLE
[4.0.6] Joomla Contact form Load Previous Form Visited

### ISSUE THREAD

**Issue by leoalv** [joomla/joomla-cms#36961]
### Steps to reproduce the issue
Create 2 categories (Uncategorized and appointment)
Create 2 or 3 custom fields of type mail and assign to Appointment category
Create 2 contacts 1 per category (general and appointment)
Create 2 menu elements of type Single Contact and select different contac in each element

### Expected result
When visit menu item for appointment load contact form with custom fields
When visit menu item for general contact load contact form without custom fields.

### Actual result
If you first visit appointment menu item it load form with custom fields, then visit General contact menu item and it also load form with custom fields

If you firts visit General contact Menu item ir load form without custom field, then visit Appointment menu and it also load form without custom fields.

### System information (as much as possible)
Joomla 4.0.6 Core, PHP 7.4.x, PHP 8, MySQL 5.7.23, Apache 2.4.46

### Additional comments
In short, when you visit the second form, the form you visited on the previous page is loaded. Only when you refresh the correct form is loaded. This behavior is not correct.

**Comment by leoalv** [comment]
See issue in action
https://user-images.githubusercontent.com/5317182/152924219-316325af-cd8c-45f7-98d0-d084a9c1820b.mp4

**Comment by brianteeman** [comment]
Very weird but I can confirm

**Comment by joomdonation** [comment]
@leoalv Could you please test PR https://github.com/joomla/joomla-cms/pull/35763 ? It should address this issue.
